src/core/Cline.ts

```diff
@@ -1310,12 +1321,31 @@ export class Cline {
 					const ruleFiles = await fs
 						.readdir(clineRulesFilePath, { withFileTypes: true, recursive: true })
 						.then((files) => files.filter((file) => file.isFile()))
-						.then((files) => files.map((file) => path.resolve(file.parentPath, file.name)))
 					const ruleFileContent = await Promise.all(
 						ruleFiles.map(async (file) => {
-							const ruleFilePath = path.resolve(clineRulesFilePath, file)
+							const ruleFilePath = path.resolve(clineRulesFilePath, file.parentPath, file.name)
```

Comment:
The way that im understanding this is that all mdc files in a project are loaded in with each request. Doesn't this defeat the purpose of MDC files? From reddit:
> .cursorrules is included with every request so you can’t really put that much info into it as most info is irrelevant for most requests.
> Project Rules (mdc) are only included when it’s relevant so you can be really detailed and specific.
So instead we should only be including mdc files if cline ie reads or writes files in a specific location, or am I thinking about this incorrectly?

Reply:
The following line checks whether the pattern specified in globs matches.
https://github.com/cline/cline/pull/2404/files#diff-b44efde712fc6775a6ce117b0de8ef3c0c3376689c98f7a9fca7ea91d40fe0f5R1338

Reply:
@saoudrizwan  Do I understand your question wrong?